### PR TITLE
GEM-48 | Add update group attachments

### PR DESCRIPTION
**Pull request by lawos98**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt [hunk 1]
```diff
@@ -49,6 +49,19 @@ class ExternalGroupController(
...
     }
 
+    @PutMapping("/{groupId}/attachments/{attachmentId}", produces = [APPLICATION_JSON_INTERNAL_VER_1])
+    @ResponseStatus(OK)
+    fun updateAttachment(
+        @RequestBody fileBytes: ByteArray,
+        @GemUserId userId: String,
+        @PathVariable groupId: String,
+        @PathVariable attachmentId: String,
+    ): GroupAttachmentResponse {
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt:62]: should update endpoint return attachmentId even though it never changes? (Just asking meybe i am wrong)

File: src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt [hunk 2]
```diff
@@ -29,11 +30,28 @@ class GroupService(
             contentType = contentType,
             sizeInBytes = size,
             file = Binary(data),
-            attachmentHistory = listOf(createNewAttachmentHistory(userId, size)),
+            attachmentHistory = listOf(createNewAttachmentHistory(userId, size, contentType)),
         )
         return groupAttachmentRepository.save(groupAttachment)
     }
 
+    fun updateAttachment(data: ByteArray, attachmentId: String, groupId: String, userId: String): GroupAttachment {
```
Comment (pwgacek) [inline src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt:39]: shouldn't we allow only Attachment author to change it? If i am not mistaken we've got only two possible scenarios for adding new group attachment: groupImage & expenseAttachment. I think we should allow only group creator to change image (as only group creator can change group name). Same case for expense attachment.

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt:39]: Will reports be stored in attachment store? If yes, i don't think we should allow to change them.

Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt:39]: what if somebody would try to update image with for example csv report

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt [hunk 3]
```diff
@@ -36,4 +36,12 @@ class ServiceTestClient(applicationContext: WebApplicationContext) {
             .uri(URI("/internal/groups/$groupId/users/$userId/generate"))
             .exchange()
     }
+
+    fun updateGroupAttachment(body: Any?, user: GemUser, attachmentId: String, groupId: String): ResponseSpec {
+        return webClient.put()
+            .uri(URI("/external/groups/$groupId/attachments/$attachmentId"))
```
Comment (pwgacek) [inline src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt]: please use EXTERNAL constant

**Review by pwgacek — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt (hunk 1 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt:62]: for my POV we should return the same dto like when we create new attachemnts soo i added that, but now we retrive only id.

File: src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt (hunk 2 above)
Comment (lawos98) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt:39]: i dont sure if we want to set that only creator of group can change that  photo or name of group. I added flat to chose what option we want to have

File: src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt (hunk 3 above)
Comment (lawos98) [reply src/integration/kotlin/pl/edu/agh/gem/integration/ability/ServiceTestClient.kt]: changed

File: src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt (hunk 1 above)
Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/external/controller/ExternalGroupController.kt:62]: ok

File: src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt (hunk 2 above)
Comment (pwgacek) [reply src/main/kotlin/pl/edu/agh/gem/internal/service/GroupService.kt:39]: will discuss on daily

**Review by pwgacek — APPROVED** [review]

**Pull request merged**